src/main/java/io/github/oliviercailloux/j_voting/Preference.java

```diff
@@ -67,23 +67,18 @@ public Alternative getAlternative(Integer position)
         return preference;
     }
 
-    /**
-     * @return the string representing a preference.
-     */
     @Override
     public String toString() {
-        LOGGER.debug("toString:");
-        String s = "";
+        StringBuilder s = new StringBuilder();
         for (Set<Alternative> set : preference) {
-            s += "{";
+            s.append("{");
             for (Alternative alter : set) {
-                s += alter.getId() + ",";
+                s.append(alter.getId() + ",");
             }
-            s = s.substring(0, s.length() - 1) + "},";
+            s.delete(s.length() - 1, s.length()).append("},");
         }
-        s = s.substring(0, s.length() - 1);
-        LOGGER.debug("preference string : {}", s);
-        return s;
+        s.delete(s.length() - 1, s.length());
+        return s.toString();
```

Comment:
Cette conversion n’est pas idéale, car le résultat n’indique aucunement qu’il s’agit d’une Préférence, c’est un simple ensemble d’ensemble d’entiers, qui pourrait représenter toutes sortes de choses.
Il aurait mieux valu utiliser Guava StringBuilder, et utiliser pour le contenu des ensembles la simple méthode existante `toString()` sur les ensembles, qui convient très bien ici (résultat moins compact mais plus précis, c’est ce qu’on veut privilégier dans `toString()`) et évite le code inélégant qui revient en arrière pour effacer les virgules en trop…
(J’admets bien sûr que le problème était déjà présent avant.)